pkg/instancetype/find/spec.go

```diff
@@ -0,0 +1,63 @@
+package find
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/tools/cache"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	api "kubevirt.io/api/instancetype"
+	"kubevirt.io/api/instancetype/v1beta1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/instancetype/compatibility"
+)
+
+type SpecFinder struct {
+	instancetypeFinder        *InstancetypeFinder
+	clusterInstancetypeFinder *ClusterInstancetypeFinder
+	revisionFinder            *RevisionFinder
+}
+
+func NewSpecFinder(store, clusterStore, revisionStore cache.Store, virtclient kubecli.KubevirtClient) *SpecFinder {
+	return &SpecFinder{
+		instancetypeFinder:        NewInstancetypeFinder(store, virtclient),
+		clusterInstancetypeFinder: NewClusterInstancetypeFinder(clusterStore, virtclient),
+		revisionFinder:            NewRevisionFinder(revisionStore, virtclient),
+	}
+}
+
+const unexpectedKindFmt = "got unexpected kind in InstancetypeMatcher: %s"
```

Comment:
Is it reused somewhere? Move into method below if not?

Reply:
I was about to use it in tests but at the moment I've not moved these alongside the extracted logic as there's an awkward import loop issue while we are still using InstancetypeMethods from the root package still.